crates/rethnet_eth/src/remote/client.rs

```diff
@@ -108,11 +111,11 @@ impl RpcClient {
             .create(directory.clone())
             .expect("failed to create on-disk RPC response cache");
 
-        Path::new(&directory).join(format!("{}.json", Self::hash_string(request_body)))
+        Path::new(&directory).join(format!("{}.json", cache_key))
     }
 
-    fn read_response_from_cache(&self, request_body: &str) -> Option<String> {
-        if let Ok(mut file) = std::fs::File::open(self.make_cache_path(request_body)) {
+    fn read_response_from_cache(&self, cache_key: Option<&CacheKey>) -> Option<String> {
```

Comment:
Why would you pass in an option to a function if it is always guaranteed to fail?

Reply:
In order to minimize conflicts with https://github.com/NomicFoundation/hardhat/pull/4270. It's fixed there.